fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -474,6 +481,126 @@ struct HasBeenTrueFor : ReferenceCounted<HasBeenTrueFor> {
 	Promise<Void> cleared;
 };
 
+std::string describeSplit(KeyRange keys, Standalone<VectorRef<KeyRef>>& splitKeys) {
+	std::string s;
+	s += "[" + keys.begin.toString() + ", " + keys.end.toString() + ") -> ";
+
+	for (auto& sk : splitKeys) {
+		s += sk.printable() + " ";
+	}
+
+	return s;
+}
+
+void executeShardSplit(DataDistributionTracker* self,
+                       KeyRange keys,
+                       Standalone<VectorRef<KeyRef>> splitKeys,
+                       Reference<AsyncVar<Optional<ShardMetrics>>> shardSize,
+                       bool relocate,
+                       RelocateReason reason) {
+
+	int numShards = splitKeys.size() - 1;
+	ASSERT(numShards > 1);
+
+	int skipRange = deterministicRandom()->randomInt(0, numShards);
+
+	auto s = describeSplit(keys, splitKeys);
+	TraceEvent(SevInfo, "ExecutingShardSplit").detail("Spltting", s).detail("NumShards", numShards);
```

Comment:
```suggestion
	TraceEvent(SevInfo, "ExecutingShardSplit").detail("Splitting", s).detail("NumShards", numShards);
```

Reply:
Also, I suggest adding a `suppressFor` here as I saw in the past there is a burst of shard splitting when the write workload is high.